packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/diagramfilter/services/DiagramFilterDescriptionProvider.java

```diff
@@ -0,0 +1,333 @@
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.diagrams.CollapsingState;
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.diagrams.Node;
+import org.eclipse.sirius.components.diagrams.ViewModifier;
+import org.eclipse.sirius.components.forms.WidgetIdProvider;
+import org.eclipse.sirius.components.forms.description.AbstractControlDescription;
+import org.eclipse.sirius.components.forms.description.CheckboxDescription;
+import org.eclipse.sirius.components.forms.description.FormDescription;
+import org.eclipse.sirius.components.forms.description.GroupDescription;
+import org.eclipse.sirius.components.forms.description.LabelDescription;
+import org.eclipse.sirius.components.forms.description.PageDescription;
+import org.eclipse.sirius.components.forms.description.SplitButtonDescription;
+import org.eclipse.sirius.components.forms.description.TreeDescription;
+import org.eclipse.sirius.components.representations.GetOrCreateRandomIdProvider;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.web.application.views.diagramfilter.services.api.IDiagramFilterActionContributionProvider;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides a tree description listing diagram elements and allowing to perform actions on them.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DiagramFilterDescriptionProvider implements IDiagramFilterDescriptionProvider {
+
+    public static final String FORM_DESCRIPTION_ID = "diagram_filter_form_description";
+
+    public static final String SELECTED_TREE_NODES = "selectedTreeNodes";
+
+    private static final String GROUP_DESCRIPTION_ID = "defaultDiagramFilterGroup";
+
+    private static final String PAGE_DESCRIPTION_ID = "defaultDiagramFilterPage";
+
+    private static final String FORM_TITLE = "Diagram Filters";
+
+    private final IObjectService objectService;
+
+    private IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    private List<IDiagramFilterActionContributionProvider> splitButtonContributionProvider;
+
+    private DiagramFilterHelper diagramFilterHelper;
+
+    public DiagramFilterDescriptionProvider(IObjectService objectService, @Lazy IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry, List<IDiagramFilterActionContributionProvider> splitButtonContributionProviders, DiagramFilterHelper diagramFilterHelper) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.editingContextEventProcessorRegistry = Objects.requireNonNull(editingContextEventProcessorRegistry);
+        this.splitButtonContributionProvider = splitButtonContributionProviders;
+        this.diagramFilterHelper = diagramFilterHelper;
```

Comment:
Some people are using parts of our code without Spring. While spring will ensure that we don't end up with an odd state, I prefer that we enforce it by ourselves too. Make your fields `final` to ensure that the state is not modified and add `Objects.requireNonNull(...)` too to tell consumers of our code that `null` is not acceptable.

This comment is valid for pretty much all other services in this package.

Reply:
Done